core/services/pipeline/common.go

```diff
@@ -172,6 +173,13 @@ func (result FinalResult) HasErrors() bool {
 	return false
 }
 
+func (result FinalResult) CombinedError() error {
+	if !result.HasErrors() {
+		return nil
+	}
+	return multierr.Combine(result.AllErrors...)
```

Comment:
maybe it make sense to use std error package join here?
```suggestion
	return errors.Join(result.AllErrors...)
```

Reply:
ooh nice, I didn't know this existed